flow/include/flow/Arena.h

```diff
@@ -672,19 +672,19 @@ inline std::string StringRef::printable() const {
 }
 
 template <class T>
-struct Traceable<Standalone<T>> : std::conditional<Traceable<T>::value, std::true_type, std::false_type>::type {
-	static std::string toString(const Standalone<T>& value) { return Traceable<T>::toString(value); }
-};
+struct Traceable<Standalone<T>> : Traceable<T> {};
 
 #define __FILE__sr StringRef(reinterpret_cast<const uint8_t*>(__FILE__), sizeof(__FILE__) - 1)
 #define __FUNCTION__sr StringRef(reinterpret_cast<const uint8_t*>(__FUNCTION__), sizeof(__FUNCTION__) - 1)
 
+template <class T>
+struct fmt::formatter<Standalone<T>> : fmt::formatter<T> {};
+
 template <>
-struct fmt::formatter<StringRef> : formatter<std::string_view> {
+struct fmt::formatter<StringRef> : formatter<std::string> {
 	template <typename FormatContext>
 	auto format(const StringRef& str, FormatContext& ctx) -> decltype(ctx.out()) {
-		std::string_view view(reinterpret_cast<const char*>(str.begin()), str.size());
-		return formatter<string_view>::format(view, ctx);
+		return formatter<std::string>::format(TraceableStringImpl<StringRef>::toString(str), ctx);
```

Comment:
Yes, and similarly we have many definitions of Traceable which just call toString() on the object.  I think we should rework these things and put some careful thought into it, but not in this PR.

Reply:
I mean, though, that rather than making the multiple changes that are done in this PR to add different formatters, could we just do one change that applies across all of them? It simplifies this PR and probably does what we would want

Reply:
If you're saying what I think you are saying, which is that any type which is supported by `Traceable<>` would be supported by `fmt::formatter<>` without writing any extra code, then I agree but I couldn't figure out how to do it.

For example this works to define `formatter<T>` for any `T` that is supported by `Traceable` via checking for `Traceable<T>::value` to be true, however it causes ambiguous specialization errors everywhere because many types such as `int` are supported explicitly by both `fmt::formatter<T>` and `Traceable<T>`.

```
template <typename T>
struct fmt::formatter<T, char, typename std::enable_if<Traceable<T>::value>::type> : formatter<std::string> {
	template <typename FormatContext>
	auto format(const T& val, FormatContext& ctx) {
		return fmt::formatter<std::string>::format(Traceable<T>::toString(val), ctx);
	}
};
```

Reply:
Or maybe you meant a generic formatter have to opt-in to?
```
template <typename T>
struct fmt_use_traceable : fmt::formatter<std::string> {

	template <typename FormatContext>
	auto format(const T& val, FormatContext& ctx) {
		return fmt::formatter<std::string>::format(Traceable<T>::toString(val), ctx);
	}
};

// Can be used as

template <typename T>
struct fmt::formatter<Optional<T>> : fmt_use_traceable<Optional<T>> {};

template <>
struct fmt::formatter<DDRangeConfig> : fmt_use_traceable<DDRangeConfig> {};
```

Reply:
This seems to work, so I've added these to the appropriate places.